web/docs/auth/Pills.jsx

```
@@ -1,99 +1,57 @@
+  discord: { label: "Discord", link: "/docs/auth/social-auth/discord" },
+  github: { label: "Github", link: "/docs/auth/social-auth/github" },
+  google: { label: "Google", link: "/docs/auth/social-auth/google" },
+  keycloak: { label: "Keycloak", link: "/docs/auth/social-auth/keycloak" },
+};
+
+function Pill({ label, link }) {
   return (
     <Link
-      to={linkToPage}
+      to={link}
       style={{
-        padding: '0.1rem 0.5rem',
-        borderRadius: '0.375rem',
-        color: 'var(--auth-pills-color)',
-        textDecoration: 'none',
-        display: 'inline-block',
-        ...style,
+        padding: "0.1rem 0.5rem",
+        borderRadius: "0.375rem",
+        color: "white",
+        backgroundColor: "#444",
+        textDecoration: "none",
+        display: "inline-block",
+        marginRight: "0.25rem",
       }}
     >
-      {children}
+      {label}
     </Link>
-  )
-}
-
-export function EmailPill() {
-  return (
-    <Pill
-      style={{
-        backgroundColor: 'var(--auth-pills-email)',
-      }}
-      linkToPage="/docs/auth/email"
-    >
-      Email
-    </Pill>
-  )
-}
-
-export function UsernameAndPasswordPill() {
-  return (
-    <Pill
-      style={{
-        backgroundColor: 'var(--auth-pills-username-and-pass)',
-      }}
-      linkToPage="/docs/auth/username-and-pass"
-    >
-      Username & Password
-    </Pill>
-  )
-}
-
-export function DiscordPill() {
-  return (
-    <Pill
-      style={{
-        backgroundColor: 'var(--auth-pills-discord)',
-      }}
-      linkToPage="/docs/auth/social-auth/discord"
-    >
-      Discord
-    </Pill>
-  )
+  );
 }
 
-export function GithubPill() {
-  return (
-    <Pill
-      style={{
-        backgroundColor: 'var(--auth-pills-github)',
-      }}
-      linkToPage="/docs/auth/social-auth/github"
-    >
-      Github
-    </Pill>
-  )
-}
+export function AuthPills({ providers }) {
+  const validPills = providers
+    .map((provider) => PILL_INFO[provider])
+    .filter(Boolean);
```

Let's make this a bit more robust and resistant to mis-use: instead of silently ignoring bad provider names, throw an error if the provider name doesn't exist!

---

added in new pr, #2821